torch/distributed/_functional_ops.py

```diff
@@ -0,0 +1,35 @@
+import torch
+import torch.distributed as dist
+
+c10_lib = torch.library.Library("c10d_functional", "DEF")
+c10_lib_impl = torch.library.Library("c10d_functional", "IMPL")
+
+ops_defs = [
+    "all_reduce(Tensor self, str reduceOp, str tag, int[] ranks, int group_size) -> Tensor",
+    "wait_tensor(Tensor self) -> Tensor",
+    "all_gather_into_tensor(Tensor shard, str tag, int[] ranks, int group_size) -> Tensor",
+    "reduce_scatter_tensor(Tensor input, str reduceOp, str tag, int[] ranks, int group_size) -> Tensor",
+]
+
+def _register_ops(resolve_ops):
```

Comment:
why is resolve_ops passed here, it's not used?

Reply:
Yeah, it's half done. Fixed :/